### ISSUE TITLE
"Missing concrete implementation" error when an implementation is available

### ISSUE THREAD

**Issue by pylaligand** [dart-lang/sdk#29429]
Here's my class:
```
import 'package:meta/meta.dart' show immutable;

import 'package:flutter/widgets.dart';

/// This class will generate an error saying an implementation for createElement
/// is missing whereas MultiChildRenderObjectWidget has one.
/// There's also a hint saying the class should be immutable.
@immutable
class Bogus extends MultiChildRenderObjectWidget {
  @override
  RenderObject createRenderObject(BuildContext context) => null;
}
```

See `MultiChildRenderObjectWidget` at https://github.com/flutter/flutter/blob/master/packages/flutter/lib/src/widgets/framework.dart#L1338: it does declare a non-abstract `createElement` method.

And yet when running the dartanalyzer CLI I get:
```
    error • Missing concrete implementation of 'RenderObjectWidget.createElement' at apps/sysui/armadillo/lib/bogus.dart:9:7 • non_abstract_class_inherits_abstract_member_one
     hint • This class inherits from a class marked as @immutable, and therefore should be immutable (all instance fields must be final) at apps/sysui/armadillo/lib/bogus.dart:9:7 • must_be_immutable
```

Interestingly, the analysis service does not complain about this file at all!!!

Dart is sync'd at 7db51c7624a26e2c4cfded9f6ea962030a6797ba (from yesterday).

@bwilkerson, who should be looking at this?

/cc @Hixie who's filed a similar issue recently, @abarth

**Comment by pylaligand** [comment]
I looked at the analysis code, and interestingly in `inheritance_manager.dart` when looking at the methods of the `MultiChildRenderObjectWidget` class neither `createElement` or the property `children` appear. Instead there seems to be just a property with an empty name.

**Comment by bwilkerson** [comment]
I have a CL out for review (https://codereview.chromium.org/2844663003/) that will partially fix the issue, but additional work is needed on the Flutter side.

tl;dr If the Flutter code base uses an optional language feature it must be enabled in all user packages. If it isn't, analyzer won't be able to analyze the Flutter code base. The bottom line is that it's problematic to ship production code that depends on experimental features.

I finally got back to this this morning. I'm now seeing this behavior in both the command-line analyzer and in analysis server. I stepped through the command-line analyzer and found where it's getting the analysis options file. It's getting it from `~/src/dart/flutter/packages/flutter/lib/analysis_options_user.yaml`, which is where my installation of Flutter is located. I ran `flutter upgrade` this morning, so this should be the current definition.

However, if you look in that file, it does not enable the initializer in asserts option. The analysis options in that file control how analyzer will analyze both user code and the flutter code base, so when analyzer finds an assert in the initializer list it doesn't know how to process it.

Please add `enableAssertInitializer: true` under the `language` key in this file. That won't fix the problem for users that have their own options file, but it should solve it for the rest.

Additionally, you might want to consider hard coding the `--enable-assert-initializers` to `flutter analyze` so that the option is always enabled regardless of what is or isn't in the analysis options file.

**Comment by pylaligand** [comment]
Thanks for looking into this! I'll send a PR for the Flutter side.

Note that in my tests, I'm using the standard dartanalyzer CLI and specifying my own options file. Does `flutter/packages/flutter/lib/analysis_options_user.yaml` still come into play in that case?

**Comment by bwilkerson** [comment]
No, it shouldn't. Not unless we fail to find the local `analysis_options.yaml` file.

I tested my changes by using `flutter create` to create a minimal flutter-based package and then created a file containing the code in your original post. Without my changes it was failing. With my code changes and the changes to the default options file it now passes (by not generating an error).

**Comment by pylaligand** [comment]
But without your change in the analyzer and with options including the proper attribute you'd still see an issue, correct?

**Comment by bwilkerson** [comment]
Correct, because when I ran the command-line analyzer it was using the path of the file being analyzed as a directory path and adding `.packages` to it, which prevented it from finding the default analysis options file.

**Comment by pylaligand** [comment]
Btw, I noticed that when the dartanalyzer CLI returns the "Missing concrete implementation bla bla" error in the case above, its return code is 0, and not != 0 as I would expect. I checked that introducing any other error changes the return code.

**Comment by bwilkerson** [comment]
The CL has landed. Please re-open this issue if it doesn't take care of the problem you were seeing. (We should create a separate issue for the return value not changing.)

**Comment by zebudummy** [comment]
@
<img width="960" alt="Annotation 2020-06-09 144503" src="https://user-images.githubusercontent.com/65569395/84204638-6ff30000-aa60-11ea-9379-c154dda20b4e.png">

**Comment by faisalmushtaq007** [comment]
@zebudummy have you found any solution to this issue???
